airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py

```diff
@@ -78,6 +78,16 @@ class CsvFormat(BaseModel):
         default=DEFAULT_FALSE_VALUES,
         description="A set of case-sensitive strings that should be interpreted as false values.",
     )
+    infer_datatypes: bool = Field(
+        title="Infer Datatypes",
+        default=False,
+        description="Whether to autogenerate the schema based the file content.",
+    )
+    infer_datatypes_legacy: bool = Field(
```

Comment:
one note here is we should try to using the word `legacy` since it isn't descriptive of what it's actually doing. granted it is a little hard since there are 4 categories of inference for legacy, so if we can't find a good name, maybe a link to the docs explaining what will happen

Reply:
Proposal:
```
class InferenceType(Enum):
    NONE = "None"
    PRIMITIVE_TYPES_ONLY = "Primitive Types Only"
    PRIMITIVE_AND_COMPLEX_TYPES = "Primitive and Complex Types"
```